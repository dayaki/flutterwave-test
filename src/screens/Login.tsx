import React, { useState, useContext } from 'react';
import { Alert, View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { Button, FormInput, PageTitleText } from '../common';
import { AppContext, saveUser } from '../store';
import { loginStyles as styles } from './styles';

const Login = () => {
  const { dispatch } = useContext(AppContext);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = () => {
    const regTest = /\S+@\S+\.\S+/;
    if (!regTest.test(email)) {
      Alert.alert('Please enter a valid email.');
      return;
    }

    if (!password) {
      Alert.alert('Password is required.');
      return;
    }
    const payload = {
      email,
      password,
    };
    dispatch(saveUser(payload));
  };

  return (
    <>
      <KeyboardAwareScrollView
        style={styles.container}
        contentContainerStyle={styles.wrapper}>
        <PageTitleText text="Hi Dayo," />
        <PageTitleText text="Welcome back" />
        <View style={[styles.form, { marginTop: 50 }]}>
          <FormInput
            placeholder="Email address"
            value={email}
            onChange={(val: string) => setEmail(val.toLowerCase())}
            keyboardType="email-address"
          />
          <FormInput
            placeholder="Password"
            value={password}
            onChange={(val: string) => setPassword(val)}
            isPassword
          />
          <Button
            text="Log in"
            buttonStyle={styles.button}
            onPress={handleLogin}
          />
        </View>
      </KeyboardAwareScrollView>
    </>
  );
};

export default Login;
