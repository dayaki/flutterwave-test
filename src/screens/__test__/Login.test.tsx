import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

test('Login screen show render as expected', () => {
  const login = renderer.create(<Login />).toJSON();
  expect(login).toMatchSnapshot();
});
