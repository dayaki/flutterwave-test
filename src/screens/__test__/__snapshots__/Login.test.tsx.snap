// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen show render as expected 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#F7F5F3",
      "flex": 1,
      "paddingHorizontal": 30,
      "paddingTop": 150,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#010101",
            "fontFamily": "NotoSans-Medium",
            "fontSize": 24,
            "lineHeight": 31,
          },
          undefined,
        ]
      }
    >
      Hi Dayo,
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#010101",
            "fontFamily": "NotoSans-Medium",
            "fontSize": 24,
            "lineHeight": 31,
          },
          undefined,
        ]
      }
    >
      Welcome back
    </Text>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "marginTop": 30,
          },
          Object {
            "marginTop": 50,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F1ECE4",
              "borderRadius": 10,
              "height": 72,
              "justifyContent": "center",
              "marginBottom": 20,
              "width": "100%",
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <TextInput
          keyboardType="email-address"
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Email address"
          placeholderTextColor="#474747"
          style={
            Array [
              Object {
                "color": "#010101",
                "fontFamily": "NotoSans-Regular",
                "fontSize": 16,
                "height": 72,
                "lineHeight": 21,
                "paddingLeft": 30,
                "width": "100%",
              },
              undefined,
            ]
          }
          textAlignVertical="top"
          value=""
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F1ECE4",
              "borderRadius": 10,
              "height": 72,
              "justifyContent": "center",
              "marginBottom": 20,
              "width": "100%",
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <TextInput
          keyboardType="default"
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#474747"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#010101",
                "fontFamily": "NotoSans-Regular",
                "fontSize": 16,
                "height": 72,
                "lineHeight": 21,
                "paddingLeft": 30,
                "width": "100%",
              },
              undefined,
            ]
          }
          textAlignVertical="top"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 30,
              "top": 24,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z"
                propList={
                  Array [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={4278255873}
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
              <RNSVGPath
                d="M21 12c-1.889 2.991-5.282 6-9 6s-7.111-3.009-9-6c2.299-2.842 4.992-6 9-6s6.701 3.158 9 6Z"
                propList={
                  Array [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={4278255873}
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DDB582",
            "borderRadius": 10,
            "height": 60,
            "justifyContent": "center",
            "marginTop": 40,
            "opacity": 1,
            "width": "100%",
          }
        }
        testID={null}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#010101",
                "fontFamily": "NotoSans-Medium",
                "fontSize": 14,
                "lineHeight": 21,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Log in
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
